packages/ui-lib/src/Filters/molecules/DropdownDatePicker.tsx

```diff
@@ -65,39 +73,61 @@ const DropdownDatePicker: React.FC<IDropdownDatePicker> = ({
 
   const handleOnToggle = useCallback((isOpen: boolean) => {
 
-    if (pickerValue.current && (pickerValue.current !== selected)) {
+    if (!hasApply && pickerValue.current && (pickerValue.current !== selected)) {
       onToggleCallback(pickerValue.current, isOpen);
     }
 
-    /** mounting again and clearing the last value sent from picker */
-    if (!isOpen && (pickerValue.current !== null) && (!mountedPicker.isMount)) {
-      pickerValue.current = null;
-      setMountedPicker({ initialValue: undefined, isMount: true });
+    /** Verify if the picker needs to be mounted again to force initialValue set in Datepicker Component */
+    if (isOpen && (!mountedPicker.isMount)) {
+      setMountedPicker((prev) => {
+        return { ...prev, isMount: true };
+      });
+    }
+  }, [hasApply, mountedPicker.isMount, onToggleCallback, selected]);
+
+  const handleOnCancel = useCallback(() => {
+    if (pickerValue.current && (pickerValue.current !== selected)) {
+      pickerValue.current = selected === undefined ? null : selected;
```

Comment:
There are a few places you have used this pattern to override `pickerValue.current`, which is directly updating that ref. My reaction is that while it works, it seems like it's not best practice... however I think this is likely an issue that's not relevant in this particular usage.

So my question here is, is there any chance of problems directly manipulating this `.current` state? Is there a better alternative pattern to this, or is this best practice?

Thanks! Happy to read any links if you've got any examples or if you want to jump on a call let me know. Aside from this sanity check and the other comment though, I think we're good!

Reply:
I think I would understand better your concerns in a call.
Will follow up on slack

Reply:
Since we have review together on chat the usage of this reference is to keep value of the selection between re-renders, I will close this comment.

The conclusion:

The reassignment of values is necessary because the `initialValue` prop in DatePicker can be undefined, but null values are not accepted. Meanwhile, pickerValue.current can be null, but it cannot be undefined.

`pickerValue` is the middle reference between the displayed value and the selected value and it helps with validation and becomes the source of truth of what to send as proper callback.